server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -429,14 +432,15 @@ protected void populateMail(Message message, MailImpl mail) throws JMSException
      * @param message The attribute source.
      * @param mail    The mail on which attribute should be set.
      * @param name    The attribute name.
+     * @throws IOException 
      */
-    private void setMailAttribute(Message message, Mail mail, String name) {
+    private void setMailAttribute(Message message, Mail mail, String name) throws IOException {
         // Now cast the property back to Serializable and set it as attribute.
         // See JAMES-1241
         Object attrValue = Throwing.function(message::getObjectProperty).apply(name);
 
         if (attrValue instanceof String) {
-            mail.setAttribute(name, SerializationUtil.deserialize((String) attrValue));
+            mail.setAttribute(new Attribute(AttributeName.of(name), AttributeValue.fromJsonString((String) attrValue)));
```

Comment:
This seems like a big change, not only JMS Mail Queue relies on the new Attribute API but also does the move to Jsonable serialization.

While the intention is noble, the risk is big and that might mess up with the community.

Note that you also need to give a migration plan for such a change: would losing all the attributes of currently enqueued emails OK?

I would advocate for stability in the JMS queues and rely on Jsonable only for new RabbitMQ stuff.